theano/tensor/tests/test_sort.py

```diff
@@ -219,3 +235,247 @@ def test_argsort_grad():
 
     data = np.random.rand(2, 3, 3).astype(theano.config.floatX)
     utt.verify_grad(lambda x: argsort(x, axis=2), [data])
+
+
+class Test_TopK(unittest.TestCase):
+
+    def setUp(self):
+        pass
+
+    @utt.parameterized.expand(product(
+        _dtypes, _int_dtypes, [-1, 0, None]))
+    def test_argtopk_sanity(self, dtype, idx_dtype, axis):
+        x = tensor.vector(name='x', dtype=dtype)
+        fn = theano.function([x], argtopk(x, 1, axis=axis, idx_dtype=idx_dtype))
+        xval = np.asarray([1]).astype(dtype)
+        yval = fn(xval)
+        assert yval == np.asarray([0], dtype=idx_dtype)
+        assert yval.dtype == np.dtype(idx_dtype)
+
+    @utt.parameterized.expand(product(
+        _dtypes, [-1, 0, None]))
+    def test_topk_sanity(self, dtype, axis):
+        x = tensor.vector(name='x', dtype=dtype)
+        fn = theano.function([x], topk(x, 1, axis=axis))
+        xval = np.asarray([1]).astype(dtype)
+        yval = fn(xval)
+        assert yval == xval
+        assert yval.dtype == xval.dtype
+
+    @utt.parameterized.expand(product(
+        _dtypes, _int_dtypes, [-1, 0, None]))
+    def test_combined_sanity(self, dtype, idx_dtype, axis):
+        x = tensor.vector(name='x', dtype=dtype)
+        yv, yi = topk_and_argtopk(x, 1, axis=axis, idx_dtype=idx_dtype)
+        fn = theano.function([x], [yv, yi])
+        xval = np.asarray([1]).astype(dtype)
+        yvval, yival = fn(xval)
+        assert yival == np.asarray([0], dtype=idx_dtype)
+        assert np.allclose(xval, yvval)
+        assert yvval.dtype == xval.dtype
+        assert yival.dtype == np.dtype(idx_dtype)
+
+    @utt.parameterized.expand(chain(
+        product(
+            (16, 61, 257),
+            (1, -1, 10, -10, 'n//2', 'n-1', '-n', '1-n'),
+            ('float64', 'float16', 'int16', 'int8')),
+        ((2049, 1337, 'float64'),)))
```

Comment:
In GPU, max size for thread block is 1024. There are two kernels on GPU, a small version for length <= 1024 and a larger version. This forces GPU to test the "large" kernel.